panimg/image_builders/dicom.py

```diff
@@ -259,7 +251,11 @@ def _process_dicom_file(*, dicom_ds):  # noqa: C901
 
     for f in OPTIONAL_METADATA_FIELDS:
         if hasattr(ref_file, f):
-            img.SetMetaData(f, str(getattr(ref_file, f)))
+            value = getattr(ref_file, f)
+            key_to_md = {md.keyword: md for md in EXTRA_METADATA}
+            if f in key_to_md:
+                key_to_md[f].validate_value(value)
```

Comment:
This construction is a little confusing, re-constructed for each item in the header and duplicated elsewhere in the code. Could it instead be moved to a method?